src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -792,12 +792,17 @@ bool async_manifest_view::in_stm(async_view_search_query_t o) {
           return ko >= sko;
       },
       [this](model::timestamp ts) {
-          auto sm = _stm_manifest.timequery(ts);
-          if (!sm.has_value()) {
-              return false;
+          vlog(_ctxlog.debug, "Checking timestamp {} using timequery", ts);
+          if (_stm_manifest.get_spillover_map().empty()) {
+              // The STM manifest is empty, so the timestamp has to be directed
+              // to the STM manifest.
+              // Implicitly, this case also handles the empty manifest case
+              // because the STM manifest with spillover segments is never
+              // empty.
+              return true;
           }
-          return sm.value().base_timestamp <= ts
-                 && ts <= sm.value().max_timestamp;
+          return _stm_manifest.get_spillover_map().last_segment()->max_timestamp
+                 < ts;
```

Comment:
how does this condition handle the handle all that had previously been handled by `partition_manifest::timequery`? What I mean is, is there a case where the timestamp isn't handled by the stm, but also isn't handled by the manifest? I guess that happens when there are no segments in the manifest or the timestamp is above the max...